packages/core-files/createFilesCollection/helpers.ts

```diff
@@ -0,0 +1,281 @@
+import fileType from 'file-type';
+import crypto from 'crypto';
+import { Meteor } from 'meteor/meteor';
+import { MongoInternals } from 'meteor/mongo';
+import { FileObj, ExtensionObj, MediaObj } from './types';
+
+export const bound = Meteor.bindEnvironment((callback) => callback());
+
+export const helpers = {
+  isString(val): boolean {
+    return val && typeof val.valueOf() === 'string';
+  },
+  isNumber(val): boolean {
+    return val && typeof val.valueOf() === 'number';
+  },
+  isUndefined(obj): boolean {
+    return obj === undefined;
+  },
+  isObject(obj): boolean {
+    if (this.isArray(obj) || this.isFunction(obj)) {
+      return false;
+    }
+    return obj === Object(obj);
+  },
+  isArray(obj): boolean {
+    return Array.isArray(obj);
+  },
+  isDate(dateStr): boolean {
+    return !Number.isNaN(new Date(dateStr).getDate());
+  },
+  isBoolean(obj): boolean {
+    return (
+      obj === true ||
+      obj === false ||
+      Object.prototype.toString.call(obj) === '[object Boolean]'
+    );
+  },
+  isFunction(obj): boolean {
+    return typeof obj === 'function' || false;
+  },
+  isEmpty(obj): boolean {
+    if (this.isDate(obj)) {
+      return false;
+    }
+    if (this.isObject(obj)) {
+      return !Object.keys(obj).length;
+    }
+    if (this.isArray(obj) || this.isString(obj)) {
+      return !obj.length;
+    }
+    return false;
+  },
+  has(_obj, path): boolean {
+    let obj = _obj;
+    if (!this.isObject(obj)) {
+      return false;
+    }
+    if (!this.isArray(path)) {
+      return (
+        this.isObject(obj) && Object.prototype.hasOwnProperty.call(obj, path)
+      );
+    }
+
+    const { length } = path;
+    for (let i = 0; i < length; i + 1) {
+      if (!Object.prototype.hasOwnProperty.call(obj, path[i])) {
+        return false;
+      }
+      obj = obj[path[i]];
+    }
+    return !!length;
+  },
+  now: Date.now,
+};
+
+export const accessDenied = (http, rc: number): boolean => {
```

Comment:
should be part of api package, see createBulkImportServer for a reference how to handle stuff at certain paths with permission checks and user context